### PR TITLE
scheduler: Refactor new API endpoints to use compojure-api.

### PR DESCRIPTION
**Pull request by mforsyth**
This also will serve to document the newer endpoints (/share, /quota,
/retry) via swagger.

### CODE REVIEW

File: scheduler/src/cook/mesos/api.clj [hunk 1]
```diff
@@ -820,19 +920,70 @@
                             :parameters {:body-params {:jobs [JobRequest]}}
                             :responses {200 {:schema [JobResponse]}}}
                      :delete {:summary "Cancels jobs, halting execution when possible."
-                              :responses {204 {:description "The jobs have been marked for termination."}
+                              :responses {204 {:description "The jobs have been markped for termination."}
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/api.clj]: markped -> marked

File: scheduler/src/cook/mesos/api.clj [hunk 2]
```diff
@@ -655,151 +698,207 @@
...
-        :delete! (fn [ctx]
-                   (quota/retract-quota! conn (get-in ctx [:request :params "user"])))
-        :handle-ok (fn [ctx]
-                     (quota/get-quota (db conn) (get-in ctx [:request :params "user"])))
-        :handle-created (fn [ctx]
-                          (quota/get-quota (db conn) (get-in ctx [:request :params "user"])))
-        :handle-accepted (fn [ctx]
-                           (quota/get-quota (db conn) (get-in ctx [:request :params "user"]))))
-      ring.middleware.json/wrap-json-params))
+  (base-retries-handler
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/api.clj]: Can you add a PUT endpoint that has the same logic? We will keep the POST for backwards compatibility but encourage users to use PUT

File: scheduler/src/cook/mesos/api.clj [hunk 3]
```diff
@@ -295,15 +322,21 @@
...
+(defn used-uuid?
+  "Returns true iff the given uuid is used in datomic"
+  [db uuid]
+  (boolean (seq (q '[:find ?j
+                     :in $ ?uuid
+                     :where
+                     [?j :job/uuid ?uuid]]
+                   db uuid))))
+
+(defn ensure-uuid-unused
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/api.clj]: :+1:

File: scheduler/src/cook/mesos/api.clj [hunk 4]
```diff
@@ -295,15 +322,21 @@
                           (conj commit-latch))))
   "ok")
 
-(defn unused-uuid?
+(defn used-uuid?
+  "Returns true iff the given uuid is used in datomic"
+  [db uuid]
+  (boolean (seq (q '[:find ?j
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/api.clj]: I think this can be simplified to:

```clojure
(-> (d/entity db [:job/uuid uuid]) nil? not)
```

**Comment by wyegelwel** [comment]
LGTM, is there anything else you plan to add before I merge?

**Comment by mforsyth** [comment]
Good to go @wyegelwel !

**Pull request merged**